### PR TITLE
Calculate sidepots

### PR DESCRIPTION
**Pull request by jonah-katz**
Screen Shot 2020-12-30 at 1 20 58 PM

### CODE REVIEW

File: nodejs/tests/table.test.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  let player_ids = {};
+  betting_rounds.forEach(a => {
+    a.forEach(b => {
+      player_ids[b.user_id] = true;
+    });
+  });
+  table.hand = {
+    players: Object.keys(player_ids).map(a => ({ user_id: a })),
+    betting_rounds,
+  };
```
Comment (jacklieblich) [inline nodejs/tests/table.test.js:15]: Don't take this personally, but I would really prefer more descriptive parameter names.

File: nodejs/tests/table.test.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        {
+          chips: 500,
+          players: ['1','3']
+        },
+      ];
+      let pot = table.calculatePots();
+      expect(pot).toEqual(expectation);
+    });
+
+    test('where 2 players go all in', () => {
```
Comment (jacklieblich) [inline nodejs/tests/table.test.js:72]: Is it implied a player went all in when they commit fewer chips than the previous player?

**Review by jacklieblich — APPROVED** [review]
Looks awesome. What about a test where someone re-raises?

**Comment by jonah-katz** [comment]
> Looks awesome. What about a test where someone re-raises?

Good call. Also, realized that the active betting round shouldn't be included in the pot. Otherwise you get weird side pots when people are yet to call..

File: nodejs/yarn-error.log [hunk 3]
```diff
@@ -0,0 +1,60 @@
+Arguments:
```
[the remark this answers is not in the document]
Comment (jonah-katz) [reply nodejs/yarn-error.log:1]: nope, good catch

File: nodejs/tests/table.test.js (hunk 1 above)
Comment (jonah-katz) [reply nodejs/tests/table.test.js:15]: Hear ya

File: nodejs/tests/table.test.js (hunk 2 above)
Comment (jonah-katz) [reply nodejs/tests/table.test.js:72]: Yes.

**Pull request merged**